tests/admin_changelist/tests.py

```diff
@@ -1741,6 +1741,44 @@ def test_modifier_allows_multiple_section(self):
             self.assertIs(c.get_property("checked"), True)
         self.assertIs(checkboxes[-1].get_property("checked"), False)
 
+    def test_selection_counter_is_synced_when_page_is_shown(self):
+        from selenium.webdriver.common.by import By
+
+        self.admin_login(username="super", password="secret")
+        self.selenium.get(self.live_server_url + reverse("admin:auth_user_changelist"))
+
+        form_id = "#changelist-form"
+        first_row_selector = "%s #result_list tbody tr:first-child" % form_id
+
+        selection_indicator = self.selenium.find_element(
+            By.CSS_SELECTOR, "%s .action-counter" % form_id
+        )
+        row_checkbox = self.selenium.find_element(
+            By.CSS_SELECTOR,
+            "%s .action-select" % first_row_selector,
+        )
+
+        # Select a row.
+        row_checkbox.click()
+        self.assertEqual(selection_indicator.text, "1 of 1 selected")
+
+        # Go to another page and get back.
+        self.selenium.get(
+            self.live_server_url + reverse("admin:admin_changelist_parent_changelist")
+        )
+        self.selenium.back()
+
+        # Ensure that the selection indicator is in sync with the selected checkboxes.
+        selection_indicator = self.selenium.find_element(
+            By.CSS_SELECTOR, "%s .action-counter" % form_id
```

Comment:
I would use f-strings as well here and as it's used twice, I would use a constant for this selector